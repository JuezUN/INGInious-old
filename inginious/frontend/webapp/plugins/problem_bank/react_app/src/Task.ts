interface Task {
  id: string;
  courseId: string;
  name: string;
  author: string;
  context: string;
  tags: string[];
}

export default Task;
