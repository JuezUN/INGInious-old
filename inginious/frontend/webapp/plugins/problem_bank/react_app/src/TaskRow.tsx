import * as React from 'react';
import Task from './Task';

export interface TaskRowProps {
  task: Task;
}

class TaskRow extends React.Component<TaskRowProps, {}> {
  public constructor(props: TaskRowProps) {
    super(props);
  }

  public render() {
    return (
      <button type="button" className="list-group-item">
        <b>{this.props.task.courseId + ' - ' + this.props.task.name}</b>
        <br />
        {this.props.task.tags.join(', ')}
      </button>
    );
  }
}

export default TaskRow;
