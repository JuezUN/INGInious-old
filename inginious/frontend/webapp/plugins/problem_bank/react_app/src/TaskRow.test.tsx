import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import TaskRow from './TaskRow';
import Task from './Task';


describe('TaskRow', () => {
  const someTask: Task = {
    id: 'someId',
    name: 'language tasks!',
    author: 'andres rondon',
    context: 'n/a',
    tags: ['cpp', 'datastructures'],
  };

  let wrapper: ShallowWrapper<{}, {}>;

  beforeEach(() => {
    wrapper = shallow(<TaskRow task={someTask}/>);
  });

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

