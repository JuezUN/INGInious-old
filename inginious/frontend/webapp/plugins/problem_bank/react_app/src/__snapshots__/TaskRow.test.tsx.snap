// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskRow renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskRow
    task={
      Object {
        "author": "andres rondon",
        "context": "n/a",
        "courseId": "Algoritmos",
        "id": "someId",
        "name": "language tasks!",
        "tags": Array [
          "cpp",
          "datastructures",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b>
          Algoritmos - language tasks!
        </b>,
        <br />,
        "cpp, datastructures",
      ],
      "className": "list-group-item",
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Algoritmos - language tasks!",
        },
        "ref": null,
        "rendered": "Algoritmos - language tasks!",
        "type": "b",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "cpp, datastructures",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b>
            Algoritmos - language tasks!
          </b>,
          <br />,
          "cpp, datastructures",
        ],
        "className": "list-group-item",
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Algoritmos - language tasks!",
          },
          "ref": null,
          "rendered": "Algoritmos - language tasks!",
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "cpp, datastructures",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
